### PR TITLE
wave13: fixes getsegment trim toward rows_filled; includes typedef

### PR DESCRIPTION
**Pull request by zack-vii**
* added function _TreeSetRowsFilled which should replace the hacky NEXT_ROW_FIX environ impl
* properly trim dimensions with treeref (_Execute) and/or windows definitions (lbound)
* remove bad code and streamlined buffer to struct transfer using put/get-int methods
* fixed WORDS_BIGENDIAN branch (compiling errors)
* fixed all main code clobbered (no-more need for pragma ignored)
* fixed stdout interference with mdsip protocol for Tunnel
* added some tests

I browsed the code and made a lot of changes:
 basically croping code and sticking it into inline statics where possible.
 some minor usage of macros to handle the methods that convert between struct and buffer.
i added some helper functions to treeshrp.h inspired by java ByteBuffer.

The PR is base on #1502 which does not pass the test for some reason. It incorporates tyedefs that will help to debug and identify argument issues, missing cases in switch statements etc.

### CODE REVIEW

**Comment by zack-vii** [comment]
retest this please

**Comment by zack-vii** [comment]
partly in wave 1 #1520  and 2 #1522

**Review by tfredian — APPROVED** [review]
